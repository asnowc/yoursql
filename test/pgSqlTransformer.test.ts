import { SqlValuesCreator, pgSqlTransformer } from "@asnc/yoursql";
import { expect, test } from "vitest";

const a = SqlValuesCreator.create(pgSqlTransformer);
test("array", function () {
  expect(a([2, 3, undefined, null, []])).toMatchSnapshot("基本转换");
  expect(a([])).toMatchSnapshot("空数组");
});
test("date", function () {
  expect(a(new Date("2022-01-13"))).toMatchSnapshot();
});
