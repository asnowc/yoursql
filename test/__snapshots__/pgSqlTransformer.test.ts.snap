// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`array > 基本转换 1`] = `"ARRAY[2,3,DEFAULT,NULL,NULL]"`;

exports[`array > 空数组 1`] = `"NULL"`;

exports[`date 1`] = `"'2022-01-13T00:00:00.000Z'"`;
